src/commands/cmd_explain.c

```diff
@@ -26,12 +26,20 @@ void Graph_Explain(void *args) {
 	QueryCtx_SetGlobalExecutionCtx(command_ctx);
 	const char *query = command_ctx->query;
 
-	// Parse the query to construct an AST
-	cypher_parse_result_t *parse_result = parse(command_ctx->query);
+	cypher_parse_result_t *params_parse_result = parse_params(command_ctx->query);
+	if(params_parse_result == NULL) goto cleanup;
+	if(AST_Validate_QueryParams(ctx, params_parse_result) != AST_VALID) goto cleanup;
+
+	// Query caching enabler.
+	AST_Extract_Params(params_parse_result);
+	const char *query_string = AST_ExtractQueryString(params_parse_result);
+
+	// Parse the query to construct an AST.
+	cypher_parse_result_t *parse_result = parse_query(query_string);
```

Comment:
Should initialize and NULL-set `parse_result` before here, as we could error on `AST_Validate_QueryParams` and jump to the cleanup routine before setting it here otherwise.